src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -59,49 +59,73 @@ class CloudSuffixNotSetException(CLIError):
 
 class CloudEndpoints:  # pylint: disable=too-few-public-methods,too-many-instance-attributes
 
+    ARM_INDEXES = {
+        "active_directory": "authentication.loginEndpoint",
+        "active_directory_data_lake_resource_id": "activeDirectoryDataLake",
+        "active_directory_graph_resource_id": "graphAudience",
+        "active_directory_resource_id": "authentication.audiences[0]",
+        "app_insights_resource_id": "appInsightsResourceId",
+        "app_insights_telemetry_channel_resource_id": "appInsightsTelemetryChannelResourceId",
+        "attestation_resource_id": "attestationResourceId",
+        "azmirror_storage_account_resource_id": "azmirrorStorageAccountResourceId",
+        "batch_resource_id": "batch",
+        "gallery": "gallery",
+        "log_analytics_resource_id": "logAnalyticsResourceId",
+        "management": "authentication.audiences[0]",
+        "media_resource_id": "media",
+        "microsoft_graph_resource_id": "microsoftGraphResourceId",
+        "ossrdbms_resource_id": "ossrdbmsResourceId",
+        "portal": "portal",
+        "resource_manager": "resourceManager",
+        "sql_management": "sqlManagement",
+        "synapse_analytics_resource_id": "synapseAnalyticsResourceId",
+        "vm_image_alias_doc": "vmImageAliasDoc"
+    }  # Please keep the endpoints in alphabetical order
```

Comment:
In Microsoft Graph's client, there is a very similar mapping:

https://github.com/Azure/azure-cli/blob/33cbd01f75c7d4c3814de5a1cbf6f42de7890467/src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_objects.py#L10-L28

Instead of using shorthand syntax, it uses a list to represent nested properties:

https://github.com/Azure/azure-cli/blob/33cbd01f75c7d4c3814de5a1cbf6f42de7890467/src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_objects.py#L20